src/expressions.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+// TODO(usergenic): Migrate this to polymer-analyzer
+// https://github.com/Polymer/polymer-analyzer/issues/351
+
+/**
+ * A parsed polymer binding expression
+ * @param {Array.<string>} keys The keys referenced by this expression.
+ * @param {Array.<string>} methods The methods referenced by this expression.
+ * @param {string} type One of 'computed', 'literal', or 'reference'
+ * @param {string} raw  The unparsed expression
+ */
+export class ParsedExpression {
```

Comment:
Recommend instead making this:

```type ParsedExpression = Computed | Literal | Reference```

And make each of those three types an interface with a string literal `type` field. That way we can separate out the fields and type safety will help us in a lot of ways.

This is how we're typing estree, and it's paid off hugely. It lets you switch over `expression.type`.. lots of stuff.